test/runtests.jl

```diff
@@ -69,6 +69,62 @@ end
                    8.848886704358202, 2.8697881579099085, 11.15107375630744, 11.8392578374927])
 end
 
+@testset "rankdeficient GLM" begin
+    # an example of rank deficiency caused by linearly dependent columns
+    num_rows = 100_000
+    dfrm = DataFrame()
+    dfrm[:x1] = randn(MersenneTwister(123), num_rows)
+    dfrm[:x2] = randn(MersenneTwister(456), num_rows)
+    dfrm[:x3]= 2*dfrm[:x1] + 3*dfrm[:x2]
+    dfrm[:y] = Int.(randn(MersenneTwister(9999999), num_rows) .> 0)
+    f1 = @eval(@formula(y ~ 1+x1+x2+x3))
+    @test_throws PosDefException fit(GeneralizedLinearModel,
+                                     f1,
+                                     dfrm,
+                                     Binomial(),
+                                     GLM.LogitLink())
+    @test_throws PosDefException fit(GeneralizedLinearModel,
+                                     f1,
+                                     dfrm,
+                                     Binomial(),
+                                     GLM.LogitLink();
+                                     allowrankdeficient=false)
+    m1 = fit(GeneralizedLinearModel,
+             f1,
+             dfrm,
+             Binomial(),
+             GLM.LogitLink();
+             allowrankdeficient=true)
+    @test isa(m1.model.pp.chol, CholeskyPivoted)
+    @test rank(m1.model.pp.chol) == 3
+    @test deviance(m1.model) ≈ 138625.6633724341
+    f2 = @eval(@formula(y ~ 1+x1*x2*x3))
+    @test_throws PosDefException fit(GeneralizedLinearModel,
+                                     f2,
+                                     dfrm,
+                                     Binomial(),
+                                     GLM.LogitLink())
+    @test_throws PosDefException fit(GeneralizedLinearModel,
+                                     f2,
+                                     dfrm,
+                                     Binomial(),
+                                     GLM.LogitLink();
+                                     allowrankdeficient=false)
+    m2 = fit(GeneralizedLinearModel,
+             f2,
+             dfrm,
+             Binomial(),
+             GLM.LogitLink();
+             allowrankdeficient=true)
+    @test isa(m2.model.pp.chol, CholeskyPivoted)
+    @test rank(m2.model.pp.chol) == 7
+    @test deviance(m2.model) ≈ 138615.90834086522
+    # issue about can't converge within 30 iterations discussed in PR #314
+    df = CSV.read(joinpath(glm_datadir, "glm.test.csv"))
```

Comment:
Can you give a more specific name to this file? For example rankdeficient_test.csv.